src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,145 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int NO_SUCH_ELEMENT_INDEX = -1;
+    private static final double MAX_CAPACITY_RAISE_INDEX = 1.5;
+
+    private T[] elementData;
+    private T[] rewriteElementData;
+    private int size = 0;
+    private int currentMaxCapacity = DEFAULT_CAPACITY;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private void resizeRewriteElementData(int newSize) {
+        rewriteElementData = (T[]) new Object[newSize];
+    }
+
+    private void resizeElementData(int newSize) {
+        elementData = (T[]) new Object[newSize];
+    }
+
+    private boolean correctIndexCheck(int index) {
```

Comment:
```suggestion
    private void checkIndex(int index) {
```
Throw exception inside